sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -74,7 +74,9 @@ case class BroadcastExchangeExec(
     child: SparkPlan) extends BroadcastExchangeLike {
   import BroadcastExchangeExec._
 
-  override val runId: UUID = UUID.randomUUID
+  private val groupId = sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID)
+
+  override val runId: UUID = Option(groupId).map(UUID.fromString).getOrElse(UUID.randomUUID)
```

Comment:
nit: we can inline `groupId` here to save a `val`, which will increase the size of the serialized plan. We should also add a comment to explain it.

Reply:
sure